wp/lib/bap_wp/src/postprocess.ml

```diff
@@ -0,0 +1,133 @@
+(***************************************************************************)
+(*                                                                         *)
+(*  Copyright (C) 2018/2019 The Charles Stark Draper Laboratory, Inc.      *)
+(*                                                                         *)
+(*  This file is provided under the license found in the LICENSE file in   *)
+(*  the top-level directory of this project.                               *)
+(*                                                                         *)
+(*  This work is funded in part by ONR/NAWC Contract N6833518C0107.  Its   *)
+(*  content does not necessarily reflect the position or policy of the US  *)
+(*  Government and no official endorsement should be inferred.             *)
+(*                                                                         *)
+(***************************************************************************)
+
+open !Core_kernel
+open Bap.Std
+
+include Self()
+
+module Expr = Z3.Expr
+module Arith = Z3.Arithmetic
+module BV = Z3.BitVector
+module Bool = Z3.Boolean
+module Array = Z3.Z3Array
+module Env = Environment
+module Constr = Constraint
+
+
+(** Implements {!Model}. *)
+
+open Core_kernel
+open Bap.Std
+
+type sign = Unsigned | Signed
+
+let string_of_sign s =
+  match s with
+  | Unsigned -> "u"
+  | Signed -> "s"
+
+exception NoValue of string
+
+let no_value_msg e m =
+  let e_str = Z3.Expr.to_string e in
+  let m_str = Z3.Model.to_string m in
+  Printf.sprintf "No value for symbol '%s' found in model: %s" e_str m_str
+
+let log ?newline:(newline="\n") msg =
+  Printf.printf "[Model] -- %s%s%!" msg newline
+
+let word_of_str s w sign =
+  let w_str = Printf.sprintf "%s:%d%s" s w (string_of_sign sign) in
+  Word.of_string w_str
+
+let z3_val_of e m =
+  match Z3.Model.eval m e true with
+  | Some v -> v
+  | None -> raise (NoValue (no_value_msg e m))
+
+let word_of_z3_bv bv =
+  let bits = Z3.BitVector.get_size (Z3.Expr.get_sort bv) in
+  let sign = Unsigned in
+  let s = Z3.Expr.to_string bv in
+  word_of_str (String.substr_replace_first s ~pattern:"#" ~with_:"0") bits sign
+
+(* get var will actually create a new variable if it can't find the one we're asking for. *)
```

Comment:
What is `get var`? Is it referring to a `get_var` function?

Reply:
Accidental personal notes left in. It was in reference to behavior in the Env module. Removed.